fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/contracttermination/LoanContractTerminationServiceImpl.java

```
@@ -0,0 +1,162 @@
+            case LOAN_CONTRACT_TERMINATION -> {
+                if (!loan.isOpen()) {
+                    final String defaultUserMessage = "Contract termination can not be applied, Loan Account is not Active.";
+                    final ApiParameterError error = ApiParameterError.generalError("error.msg.loan.account.is.not.active.state",
+                            defaultUserMessage);
+                    dataValidationErrors.add(error);
+                }
+
+                if (loan.getLoanProduct().getLoanProductRelatedDetail().getLoanScheduleType().equals(LoanScheduleType.CUMULATIVE)) {
+                    final String defaultUserMessage = "Contract termination can not be applied, Loan product schedule type is not Progressive.";
+                    final ApiParameterError error = ApiParameterError.generalError(
+                            "error.msg.loan.contract.termination.is.only.supported.for.progressive.loan.schedule.type", defaultUserMessage);
+                    dataValidationErrors.add(error);
+                }
+
+                if (loan.isChargedOff()) {
+                    final String defaultUserMessage = "Contract termination can not be applied, Loan Account is Charge-Off.";
+                    final ApiParameterError error = ApiParameterError.generalError("error.msg.loan.account.is.charge-off",
+                            defaultUserMessage);
+                    dataValidationErrors.add(error);
+                }
+
+                if (loan.isContractTermination()) {
+                    final String defaultUserMessage = "Contract termination can not be applied, Loan Account is already terminated.";
+                    final ApiParameterError error = ApiParameterError
+                            .generalError("error.msg.loan.account.is.already.contract.termination.substate", defaultUserMessage);
+                    dataValidationErrors.add(error);
+                }
+            }
+            default -> {
+            }
```

Since this method is public, i think we should at least throw an exception in the default case.

---

I would rather say we dont event need the whole switch :)

---

Done